### PR TITLE
ES reindex Airflow workflow

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes :

This PR is done on top of https://github.com/open-metadata/OpenMetadata/pull/8679. Don't merge this until https://github.com/open-metadata/OpenMetadata/pull/8679 is in `main`

#
### Type of change :

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-spec/src/main/resources/json/schema/entity/services/connections/metadata/openMetadataConnection.json [hunk 1]
```diff
@@ -203,7 +203,7 @@
       "additionalProperties": false,
       "required": ["type"]
     },
-    "supportMetadataToElasticSearchExtraction": {
+    "supportsDataInsightExtraction": {
```
Comment (pmbrull) [inline openmetadata-spec/src/main/resources/json/schema/entity/services/connections/metadata/openMetadataConnection.json:206]: naming was not correct

File: openmetadata-spec/src/main/resources/json/schema/metadataIngestion/metadataToElasticSearchPipeline.json [hunk 2]
```diff
@@ -18,12 +18,14 @@
       "default": "MetadataToElasticSearch"
     },
     "useSSL": {
-      "description": "Indicates whether to use SSL",
+      "title": "Use SSL",
+      "description": "Indicates whether to use SSL when connecting to ElasticSearch. By default, we will ignore SSL settings.",
```
Comment (pmbrull) [inline openmetadata-spec/src/main/resources/json/schema/metadataIngestion/metadataToElasticSearchPipeline.json:22]: adding descriptions and titles for the UI to render

File: openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx [hunk 3]
```diff
@@ -394,6 +394,8 @@ const AddIngestion = ({
       }
     } else if (showDBTConfig && activeIngestionStep === 2) {
       nextStep = activeIngestionStep + 2;
+    } else if (showDBTConfig && activeIngestionStep === 2) {
```
Comment (pmbrull) [inline openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx:397]: @chirag-madlani got these changes from the UI after merging https://github.com/open-metadata/OpenMetadata/pull/8679 in this branch and merging `main`. IDK if this is correct. Please validate

File: openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/data_insight.py [hunk 4]
```diff
@@ -39,7 +39,8 @@ def build_data_insight_workflow_config(
 
     workflow_config = OpenMetadataWorkflowConfig(
         source=build_source(ingestion_pipeline),
-        sink=ingestion_pipeline.sink,
+        # ingestion_pipeline.service.serviceConnection.elasticsearch
```
Comment (nahuelverdugo) [inline openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/data_insight.py:42]:
```suggestion
```

File: openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx (hunk 3 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx:397]: @pmbrull looks like the else if condition in both `handlePrev` and `handleNext` functions got repeated while merging. If you can remove the duplication that would be great

File: openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx [hunk 5]
```diff
@@ -415,6 +417,8 @@ const AddIngestion = ({
       prevStep = activeIngestionStep - 2;
     } else if (showDBTConfig && activeIngestionStep === 4) {
       prevStep = activeIngestionStep - 2;
+    } else if (showDBTConfig && activeIngestionStep === 4) {
+      prevStep = activeIngestionStep - 2;
```
Comment (nahuelverdugo) [inline openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx:421]: is this duplicated?

File: openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx [hunk 6]
```diff
@@ -394,6 +394,8 @@ const AddIngestion = ({
       }
     } else if (showDBTConfig && activeIngestionStep === 2) {
       nextStep = activeIngestionStep + 2;
+    } else if (showDBTConfig && activeIngestionStep === 2) {
+      nextStep = activeIngestionStep + 2;
```
Comment (nahuelverdugo) [inline openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx:398]: duplicated with the previous else-if condition?

**Review by nahuelverdugo — DISMISSED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx (hunk 5 above)
Comment (pmbrull) [reply openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx:421]: yup some lingering UI code from merging a previous branch. @chirag-madlani will chip in. Thanks

File: openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx (hunk 6 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx]: Done

Comment (pmbrull) [reply openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx]: thanks folks

File: openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx (hunk 5 above)
Comment (pmbrull) [reply openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx]: thanks

File: openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/data_insight.py (hunk 4 above)
Comment (pmbrull) [reply openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/data_insight.py:42]: will be handled in https://github.com/open-metadata/OpenMetadata/pull/8729

File: openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx (hunk 3 above)
Comment (pmbrull) [reply openmetadata-ui/src/main/resources/ui/src/components/AddIngestion/AddIngestion.component.tsx]: thanks, solved!

**Review by TeddyCr — APPROVED** [review]

**Pull request merged**